Update Readme with instructions how to develop Helix SDK

premun opened the pull request:
I was giving these to way too many people lately so putting them in a more visible place

tkapin commented on src/Microsoft.DotNet.Helix/Sdk/Readme.md:
```diff
@@ -36,6 +38,49 @@ BUILD_REASON
+    export SYSTEM_TEAMPROJECT=dnceng
+    export SYSTEM_ACCESSTOKEN=''
+
+    eng/common/build.sh -test -projects tests/UnitTests.XHarness.Android.Device.proj /v:n /bl:Arcade.binlog
```
As discussed, XHarness.Android requires some additional parameters. The android / iOS examples are nice, but since this is page about Helix.SDK, shouldn't we use some simpler job (e.g., HelloWorld) to illustrate the functionality to reduce the complexity of logs etc.?

premun commented:
@tkapin I have removed the need to specify the extra properties when running Android tests and I have added a human readable error in case someone still missuses one of the testapk projects
